app/services/search_form_param_parser.rb

```diff
@@ -24,11 +27,11 @@ def parse_filters(filters)
   end
 
   def blocks_hash(filters)
-    filters.map do |id, values|
+    filters.reject{ |_,v| v == IGNORED_OPTION }.map do |id, values|
       if values.is_a?(Array)
         values.map { |value| identifier_hash(id, value) }
       else
-        identifier_hash(id, values)
+        identifier_hash(id, years_array(values))
       end
     end
   end
```

Comment:
With this PR the if statement becomes confuse to me. Is this "else" clause exists only for the year of publication field? If so can we make something like?

```
  if year_of_publication?
     convert_params(id, years_array(values))
  else
     values.map { |value| convert_params(id, value) }
  end
```
@willhall88 what do you think?